crates/sui-core/src/consensus_handler.rs

```diff
@@ -62,6 +67,15 @@ impl ExecutionState for ConsensusHandler {
         serialized_transaction: Vec<u8>,
     ) {
         let consensus_index = self.update_hash(consensus_index, &serialized_transaction);
+        let consensus_index = if let Some(consensus_index) = consensus_index {
+            consensus_index
+        } else {
+            debug!(
+                "Ignore consensus transaction at index {:?} as it appear to be already processed",
+                consensus_index
+            );
+            return;
+        };
```

Comment:
This check here seems great and very fast way to check an already execution. My only concern is that we now have 2 places where we check the same thing (have we executed this transaction?) - the second being here https://github.com/MystenLabs/sui/blob/63b92baaa3fd94695159edf497fa46e648c02a75/crates/sui-core/src/authority.rs#L2227

Do we need now the second check? If yes, then do you have any insight on what are the criteria for those 2 checks to diverge?